packages/compiler/src/core/program.ts

```diff
@@ -910,3 +922,41 @@ function resolveOptions(options: CompilerOptions): CompilerOptions {
     outputPath: outputDir,
   };
 }
+
+async function emit(emitter: EmitterRef, options: CompilerOptions = {}, program: Program) {
+  const emitterName = emitter.metadata.name ?? "";
+  const outputRelativePath = options.listFiles
+    ? `./${getRelativePathFromDirectory(program.projectRoot, emitter.emitterOutputDir, false)}/`
+    : "";
+  addChildLog(`✓ ${emitterName}\t${outputRelativePath}`);
+  setChildLogPrefix(outputRelativePath);
+
+  const logger = createLogger({ sink: program.host.logSink });
+  if (logger.trackAction) {
+    await logger.trackAction(() => runEmitter(emitter, options, program), {
+      level: "trace",
+      message: emitterName,
```

Comment:
Can we log something like `run ${emitterName}` and log-level is `info`?

Typespec extension may rely on the log to figure out which emitter is triggered and which is skipped. If it is tracing log, the log will not be sent by default.

And can we configure to add more tracking logs?

Reply:
the output of the cli is not part of the contract(and should not be) if you try to parse it it will break in the future.

That whole generate from the IDE is a mess that I didn't want to merge but yet it was. Calling the CLI should be done as it is an be done if you want to cusomize things you have to use the progrmatic api.